src/internals/helpers.ts

```diff
@@ -90,9 +89,7 @@ export const matchPattern = (
           endPatterns.every((subPattern, i) =>
             matchPattern(subPattern, endValues[i], select)
           ) &&
-          (variadicPatterns.length === 0
-            ? true
-            : matchPattern(variadicPatterns[0], middleValues, select))
+          matchPattern(variadicPatterns[0], middleValues, select)
```

Comment:
This is incorrect and would likely break test on tuple pattern.

Reply:
above check `if (variadicPatterns.length) {`